registry/registry.go

```diff
@@ -135,10 +135,27 @@ func (registry *Registry) ListenAndServe() error {
 	}
 
 	if config.HTTP.TLS.Certificate != "" || config.HTTP.TLS.LetsEncrypt.CacheFile != "" {
+		var tlsMinVersion uint16
+		var tlsString string
+		switch config.HTTP.TLS.MinimumTLS {
+		case "tls1.0":
+			tlsMinVersion = tls.VersionTLS10
+			tlsString = config.HTTP.TLS.MinimumTLS
+		case "tls1.1":
+			tlsMinVersion = tls.VersionTLS11
+			tlsString = config.HTTP.TLS.MinimumTLS
+		case "tls1.2":
+			tlsMinVersion = tls.VersionTLS12
+			tlsString = config.HTTP.TLS.MinimumTLS
+		default:
+			tlsMinVersion = tls.VersionTLS10
```

Comment:
I am concerned about silently choosing a version due to misconfiguration. Ideally the app should fail if particular config is incorrect. At minimum there should be a warning here.

Reply:
+1 on that. Also, before merging let's seriously consider dropping support for 1.{0, 1} altogether. 1.2 is 10 yrs old at this point so pretty pervasive, and our selected cipher suites (all AES) aren't even in the [1.0 spec](https://tools.ietf.org/html/rfc2246)

Reply:
And for that matter ECDHE for key exchange post-dates [1.1](https://www.ietf.org/rfc/rfc4346.txt)

Reply:
Sorry, talked myself out of a warning and went for the panic/fail in commit 111da2bd7afa564550020d9d10e732e1966549a5.  If the user went to the trouble of asking for a minimum TLS version, but made a typo, they'd probably rather have an obvious failure than running an insecure protocol and a warning they might have missed.  Omitted settings are still silently kept to the default behavior.  Code got tighter and easier to read too.  https://github.com/docker/distribution/blob/111da2bd7afa564550020d9d10e732e1966549a5/registry/registry.go#L137-L153 

Unfortunately travis failed on an unrelated 'apt update -q' ... do you want me to close and reopen the PR to trigger a build?